cmd/deploy.go

```diff
@@ -180,7 +180,7 @@ func autofillApimodel(dc *deployCmd) {
 
 	if !useManagedIdentity {
 		spp := dc.containerService.Properties.ServicePrincipalProfile
-		if spp != nil && (spp.ClientID == "" || spp.Secret == "") {
+		if spp != nil && spp.KeyvaultSecretRef == "" && (spp.ClientID == "" || spp.Secret == "") {
```

Comment:
Conceptually spp.Secert can now be found in two places. So places that used to check for its existence now need to check in two places. If only the secret ref was specified the code here will think every thing is peachy and not fail the request until something fails later since the client id is empty. If we just change this line then we will come in and generate a new SP/secret not realizing the user had tried to set the secret via KV. and both will be set which should cause issues elsewhere in the code.